addons/calendar/models/calendar_attendee.py

```diff
@@ -110,7 +110,7 @@ def _send_mail_to_attendees(self, mail_template, force_send=False):
         ics_files = self.mapped('event_id')._get_ics_file()
 
         for attendee in self:
-            if attendee.email and attendee.partner_id != self.env.user.partner_id:
```

Comment:
Hello @jeh-odoo 

Approach is looking good but I would slightly change it, to:
`if attendee.email and (attendee.partner_id != self.env.user.partner_id or attendee._is_self_notify())`

And in the method simply return False here and return self.event_id.appointment_id in the override.
Does that make sense?

Thanks & Cheers!

Reply:
I would include the partner check in the method to allow override the whole condition, like
```
        if attendee.email and self._should_notify_attendee(attendee):
            # ...

    def _should_notify_attendee(self, attendee):
        return attendee.partner_id !=self.env.user.partner_id
```
And in appointment
```
    def _should_notify_attendee(self, attendee):
        return self.event_id.appointment_type_id or super()._should_notify_attendee(attendee)
```

That way we avoid a void method, and the condition is contained in a single method instead of partly hardcoded, partly overridden.